matter_server/server/device_controller.py

```diff
@@ -168,8 +168,23 @@ async def commission_with_code(self, code: str) -> MatterNodeData:
                 f"Commission with code failed for node {node_id}"
             )
 
-        # full interview of the device
-        await self.interview_node(node_id)
+        # perform full (first) interview of the device
+        # we retry the interview max 3 times as it may fail in noisy
+        # RF environments (in case of thread), mdns trouble or just flaky devices.
+        # retryuing both the mdns resolve and (first) interview, increases the chances
```

Comment:
small typo in comment
```suggestion
        # retrying both the mdns resolve and (first) interview, increases the chances
```